### ISSUE TITLE
Make it more independent of project structure

### ISSUE THREAD

**Issue by ales-tsurko** [tauri-apps/tauri#878]
**Is your feature request related to a problem? Please describe.**

As far as I understand, Tauri can't be used without `src-tauri` directory. It's supposed to be an extension of a JS project. As for Rust-based projects, you can't omit `src-tauri` directory and should put everything related to Rust there.

For example, you already have a Cargo workspace with a structure like:
```
gui/          # JS-based GUI
some_crate_A/ # a crate, which is a part of the workspace
some_crate_B/ # another crate, which is a part of the workspace
src/          # Rust main crate, application's logic
Cargo.toml
```

But If you want to integrate Tauri into this project, you should change your project structure to something like:
```
src/              # JS-based GUI
src-tauri/                   
    some_crate_A/ # a crate, which is a part of the workspace
    some_crate_B/ # another crate, which is a part of the workspace
    src/          # Rust main crate, application's logic
    Cargo.toml
    tauri.conf.json
package.json
```

This makes the structure inconvenient for build/CI/CD scripts because you need to either change your project build tool to a JS one or add additional script at the root, or document everywhere that `src-tauri` should be considered the root and scripts should be run from there.

**Describe the solution you'd like**

Add an option to Tauri CLI to initialize a project in an existing directory. Also, it's worth providing instructions how to make it manually because the Cargo.toml and other files with same names as in the Tauri template could already exist in the project.

So you would, for example, just add `tauri.conf.json` to your project to make it possible to run `tauri dev` from its root and consider it a Tauri project. The described previously project structure would look like this after the change:

```
gui/          # JS-based GUI
some_crate_A/ # a crate, which is a part of the workspace
some_crate_B/ # another crate, which is a part of the workspace
src/          # Tauri Rust code, application's logic
Cargo.toml
tauri.conf.json
```

**Describe alternatives you've considered**
None.

**Additional context**
None.

**Comment by chippers** [comment]
> Describe alternatives you've considered
None.

I think the default of belonging to the crate the `tauri.conf.json` is in is a good default.  I think an additional field in the config file to set override the app directory could be useful too so that more project layouts are possible.  Something like `tauri.conf.json > build > appDir` or w/e you wish to name the key.

**Comment by nothingismagick** [comment]
We had a conversation about this yesterday at discord, and I think that this project layout is good alternative that will rise in popularity once we also make the rust-based CLI available...

**Comment by boehs** [comment]
Can this issue be closed? Tried tauri a while ago and it frustrated me to have that lockin. Has this been fixed?

**Comment by boehs** [comment]
Closed by #2654

**Comment by lucasfernog** [comment]
Thanks for catching this issue @boehs 

Also related to #3035 and more improvements coming soon.
